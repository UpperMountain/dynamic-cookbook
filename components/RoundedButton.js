import React from "react";
import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
import { LinearGradient } from "expo";

const styles = ({ background, borderColor, textColor }, width, size) =>
  StyleSheet.create({
    button: {
      width: width,
      height: 56,
      backgroundColor: background || "white",
      borderColor: borderColor || "black",
      borderWidth: 3,
      borderRadius: 72,
      flexDirection: "row",
      alignItems: "center",
      justifyContent: "center"
    },
    text: {
      fontSize: size,
      color: textColor || "black",
      lineHeight: 32,
      fontFamily: "raleway-bold",
      flexDirection: "row"
    }
  });

const RoundedButton = props => {
  let currentStyle = null;
  if (props.clear) {
    currentStyle = styles(
      {
        background: "transparent",
        textColor: "white",
        borderColor: "white"
      },
      props.width || 315,
      props.size || 18
    );
  } else if (props.gradient) {
    currentStyle = styles(
      {
        background: "transparent",
        textColor: "white",
        borderColor: "transparent"
      },
      props.width || 315,
      props.size || 18
    );
  } else {
    currentStyle = styles(
      props.style || {},
      props.width || 315,
      props.size || 18
    );
  }

  let buttonText = <Text style={currentStyle.text}>{props.text}</Text>;
  let outerButton = null;

  if (props.gradient) {
    outerButton = (
      <TouchableOpacity onPress={props.onPress}>
        <LinearGradient
          style={currentStyle.button}
          colors={["#FFAFBD", "#FFC3A0"]}
        >
          {buttonText}
        </LinearGradient>
      </TouchableOpacity>
    );
  } else {
    outerButton = (
      <TouchableOpacity style={currentStyle.button} onPress={props.onPress}>
        {buttonText}
      </TouchableOpacity>
    );
  }
  return <View>{outerButton}</View>;
};

export default RoundedButton;
