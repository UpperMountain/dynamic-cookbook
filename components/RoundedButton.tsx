import React from "react";
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  ViewStyle,
  TextStyle
} from "react-native";
import { LinearGradient } from "expo";

interface CustomStyle {
  background?: ViewStyle["backgroundColor"];
  borderColor?: ViewStyle["borderColor"];
  textColor?: TextStyle["color"];
}
const styles = (
  { background, borderColor, textColor }: CustomStyle,
  width: number,
  size: number
) =>
  StyleSheet.create({
    button: {
      width,
      height: 56,
      backgroundColor: background || "white",
      borderColor: borderColor || "black",
      borderWidth: 3,
      borderRadius: 72,
      flexDirection: "row",
      alignItems: "center",
      justifyContent: "center"
    },
    text: {
      fontSize: size,
      color: textColor || "black",
      lineHeight: 32,
      fontFamily: "raleway-bold",
      flexDirection: "row"
    }
  });

interface Props {
  width?: number;
  size?: number;
  style?: CustomStyle;
  clear?: boolean;
  gradient?: boolean;
  text: string;
  onPress: () => void;
}

const RoundedButton = ({
  width,
  size,
  style,
  clear,
  gradient,
  text,
  onPress
}: Props) => {
  let currentStyle = null;
  if (clear) {
    currentStyle = styles(
      {
        background: "transparent",
        textColor: "white",
        borderColor: "white"
      },
      width || 315,
      size || 18
    );
  } else if (gradient) {
    currentStyle = styles(
      {
        background: "transparent",
        textColor: "white",
        borderColor: "transparent"
      },
      width || 315,
      size || 18
    );
  } else {
    currentStyle = styles(style || {}, width || 315, size || 18);
  }

  const buttonText = <Text style={currentStyle.text}>{text}</Text>;
  let outerButton = null;

  if (gradient) {
    outerButton = (
      <TouchableOpacity onPress={onPress}>
        <LinearGradient
          style={currentStyle.button}
          colors={["#FFAFBD", "#FFC3A0"]}
        >
          {buttonText}
        </LinearGradient>
      </TouchableOpacity>
    );
  } else {
    outerButton = (
      <TouchableOpacity style={currentStyle.button} onPress={onPress}>
        {buttonText}
      </TouchableOpacity>
    );
  }
  return <View>{outerButton}</View>;
};

export default RoundedButton;
