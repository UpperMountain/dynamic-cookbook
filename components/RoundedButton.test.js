import React from "react";
import RoundedButton from "./RoundedButton";
import renderer from "react-test-renderer";

it("should render something without exploding", () => {
  const cmp = renderer.create(<RoundedButton text="test" />);
});

it("should render a plain button", () => {
  const cmp = renderer.create(<RoundedButton text="test" />);
  expect(cmp.toJSON()).toMatchSnapshot();
});
