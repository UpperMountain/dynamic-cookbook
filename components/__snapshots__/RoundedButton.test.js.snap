// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a plain button 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "black",
        "borderRadius": 72,
        "borderWidth": 3,
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "center",
        "opacity": 1,
        "width": 315,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "flexDirection": "row",
          "fontFamily": "raleway-bold",
          "fontSize": 18,
          "lineHeight": 32,
        }
      }
    >
      test
    </Text>
  </View>
</View>
`;
